semana10/astrotrips/src/actions/trips.js

```diff
@@ -0,0 +1,131 @@
+import axios from "axios"
+import { push,replace } from "connected-react-router";
+import { routes } from "../containers/Router/index";
+import moment from 'moment'
+
+export const setTrips = (trip) => {
+    return {
+        type: "SET_TRIPS",
+        payload: {
+            trip
+        }
+    }
+}
+
+export const setTripsDetails = (tripDetails) => {
+    return {
+        type: "SET_TRIP_DETAILS",
+        payload: {
+            tripDetails
+        }
+    }
+}
+
+
+export const getTrips = () => async (dispatch) => {
+    try {
+        const response = await axios.get(
+            "https://us-central1-missao-newton.cloudfunctions.net/futureX/natalia-acedo/trips"
+        )
+        dispatch(setTrips(response.data.trips))
+    } catch (error) {
+        console.error(error)
+    } 
+}
+
+
+export const applyToTrip = (form, id) => async () => {
+    try {
+        await axios.post(
+        `https://us-central1-missao-newton.cloudfunctions.net/futureX/natalia-acedo/trips/${id}/apply`,
+        form
+        )
+        alert("Aplicação enviada!")
+    } catch (error) {
+        console.error(error)
+    }
+}
+
+
+export const login = (email, password) => async (dispatch) => {
+    const body = {
+        email,
+        password
+    }
+
+    try {
+        const response = await axios.post(
+            "https://us-central1-missao-newton.cloudfunctions.net/futureX/natalia-acedo/login",
+            body
+        )
+        localStorage.setItem("token", response.data.token)
+        dispatch(push(routes.user))
+    } catch (error) {
+        console.error(error)
+        alert("Dados incorretos")
+    }
+}
+
+const token = localStorage.getItem("token");
```

Comment:
Cuidado com isso! Se o usuário começa a interagir com o site antes de fazer login, esse `token` vai ficar como `null` mesmo depois de um login feito com sucesso, para evitar esse problema, 2 sugestões:

1) Sempre que for usar, acessar diretamente do LocalStorage:
```js
export const getTripDetails = (id) => async (dispatch) => {
   const token = localStorage.getItem("token");
...
```

2) Transformar o `token` em uma função chamada `getToken`, dessa forma, todas as vezes que ela for executada, retornará o valor que está **atualmente** no LocalStorage:

```js
 const getToken = () => localStorage.getItem("token")
```

e usaria:
```js
{
  headers: {
    auth: getToken(),
  }
}
```